java/vector/src/main/java/org/apache/arrow/vector/ValueVector.java

```diff
@@ -283,4 +283,10 @@
    * @return the name of the vector.
    */
   String getName();
+
+  /**
+   * Validate the vector, will throw exception if validate fail.
+   */
+  void validate();
```

Comment:
Do we really have to add a new API in the interface? It seems the existing visitor interface is sufficient to handle it.

Reply:
Thanks for your comments, this is consistent with C ++, and I think it could make code more concise for users. If we remove this, i.e., several cases would looks like this:

if we want to do something like validateFull, one need to write code:
>VectorValidateVisitor visitor1 = new VectorValidateVisitor ();
>VectorDataValidateVisitor visitor2 = new VectorDateValidateVisitor();
>v.accept(visitor1);
>v.accept(visitor2);

and for ExtentionTypeVector, we need to get underlying vector and call visitor API then
>VectorValidateVisitor visitor = new VectorValidateVisitor ();
>v.getUderlyingVector.accept(visitor, null);

Cases above could be can be simplified to v.validate() or v.validateFull() with the two API.
And I don't insist on this if they are not needed.

Reply:
I agree with you that it makes the client code simpler. 

IMO, the real benefits are 1) one fewer virtual function call; and 2) one fewer stack frame. However, the cost is that it makes the interface larger and difficult to maintain. In addition, adding new APIs to a public interface has the risk of crashing client code. 

So I think the method of adding new APIs should be reserved for performance-critical operations.

Reply:
Instead of adding this directly to ValueVector, can we make a static utility method that does validation and excepts a ValueVector?

Reply:
Thanks for all your comments, we already have validate method in ValueVectorUtility, and I removed the api from ValueVector.